lib/ramble/ramble/application.py

```diff
@@ -803,6 +803,12 @@ def _set_default_experiment_variables(self):
                 action = vals['action']
                 value = vals['value']
 
+                # Since the type coming from the schema can either be None, or
+                # a complex polymorphic type we need to ensure it has a
+                # sensible base structure when it is not given
+                if not self._env_variable_sets:
+                    self._env_variable_sets.append({'set': {}})
```

Comment:
Is there a good reason to not also define empty dicts for `append`, `prepend`, and `unset` other than those being unsupported methods of defining variables in application.py files?

Reply:
Yea simply that we chose not to do it before, and it's not needed here